### PR TITLE
Simplify Gnosis Safe state

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

This PR takes the Gnosis Safe updater, and since it has to do with the wallets, moves it to the wallets module.

## WalletUpdater
In the `wallets` module, we have now a `WalletUpdater`. This PR integrates it there.

## Remove safe information from WalletDetails
This PR also uses one specific hook just for the safe.

The reason, is that Gnosis Safe information was not needed most of the times that we used `useWalletDetails` hook. 
So I just removed it from there.

So I now keep a separation on those. To get the info from the safe you can use a hook. 

## SDK info hook
There was a hook that was giving us the data from the Safe SDK. 
This was created just to get the `isReadOnly` flag. 

I included this information as part of the general information from the new `useGnosisSafeInfo` hook. 
I left the original hook, but just as a helper hook that is now ONLY INTERNAL to the `wallets` module. 

Now to read the `isReadOnly` we just use the normal safe hook.

## Test
- It would be nice to do a test of a swap using gnosis safe.
- @elena-zh as the PR on top of the waterfall. Would be nice to validate I didn't break any connection, so nice to test the wallets overall. Just make sure 1 injected works, safe works, and 1 wallet connect works i would think is enough

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @anxolin , the build failed. Could you please check it?

File: src/cow-react/modules/wallet/api/state.ts [hunk 1]
```diff
@@ -14,7 +14,6 @@ export const walletDetailsAtom = atom<WalletDetails>({
 
   // Feature Support
   allowsOffchainSigning: false,
-
-  // Additioal information
-  gnosisSafeInfo: undefined,
```
Comment (anxolin) [inline src/cow-react/modules/wallet/api/state.ts:19]: Now is gone from the main `WalletDetails`

File: src/cow-react/modules/application/containers/App/Updaters.tsx [hunk 2]
```diff
@@ -46,7 +45,6 @@ export function Updaters() {
       <SentryUpdater />
       <UploadToIpfsUpdater />
       <AppDataUpdater />
-      <GnosisSafeUpdater />
```
Comment (anxolin) [inline src/cow-react/modules/application/containers/App/Updaters.tsx:49]: we don't need a specific updater for the safe. The `WalletUpdater` will include it in its responsability

File: src/cow-react/modules/wallet/api/hooks.ts [hunk 3]
```diff
@@ -9,3 +9,7 @@ export function useWalletInfo(): WalletInfo {
 export function useWalletDetails(): WalletDetails {
   return useAtomValue(walletDetailsAtom)
 }
+
+export function useGnosisSafeInfo(): GnosisSafeInfo | undefined {
+  return useAtomValue(gnosisSafeInfoAtom)
+}
```
Comment (anxolin) [inline src/cow-react/modules/wallet/api/hooks.ts:15]: the atom now is now moved to the generic hooks (they don't depend from any specific implementation, it just read from joitai state)

File: src/cow-react/modules/wallet/web3-react/hooks/useGnosisSafeSdkInfo.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react'
+
+import { GnosisSafe } from '@web3-react/gnosis-safe'
+import { Connector } from '@web3-react/types'
+import type { SafeInfo } from '@gnosis.pm/safe-apps-sdk'
+
+export type GnosisSafeSdkInfo = SafeInfo
+
+export function useGnosisSafeSdkInfo(connector: Connector, active?: boolean): GnosisSafeSdkInfo | null {
```
Comment (anxolin) [inline src/cow-react/modules/wallet/web3-react/hooks/useGnosisSafeSdkInfo.ts:9]: This hook is similar to the one called before `useGnosisSafeInfo`. 

I added the word `Sdk` to make sure we understand this hook only applies to that provider/connector.


This hook is not exposed out of the `wallet` module any more. Now is used just as a helper hook for the updater

**Review by elena-zh — COMMENTED** [review]
Hey @anxolin , everything works great!

The same issues that I reported in https://github.com/cowprotocol/cowswap/pull/2101#pullrequestreview-1326439758 are reproducible here as well. 

I faced only 1 issue that would be great to address here: it is impossible to disconnect a WC wallet on a mobile device now (it is not reproducible in Dev and in #2101 PR): 
1. open the PR preview link on a mobile device, 
2. connect to any wallet using WC there. 
3. Then disconnect it. 
4. Refresh the page OR press on the Connect wallet once again --->  you will see that you are connected to the previous wallet. However, a user might want to connect to another ono, so it would be great to kill the previous connection. 

See the video with this issue: 

https://user-images.githubusercontent.com/70885163/223171155-2c9486fc-c6ee-4553-a7da-3b0ef74fe1d3.mp4

Thanks!

**Comment by anxolin** [comment]
> Refresh the page OR press on the Connect wallet once again ---> you will see that you are connected to the previous wallet. However, a user might want to connect to another ono, so it would be great to kill the previous connection.

I'm puzzled by this. I've reviewed the code, and I don't see where this might be coming from, since here I changed the code for Gnosis Safe, nothing for Wallet Connect. 

Are you sure this was not just a once time thing. It's possible you tried to refresh too fast, before the connection was killed?

**Comment by elena-zh** [comment]
Hey @anxolin , I have retested it today. 
Not reproducible for me today. It might have been a glitch ;(

Again, sorry about the noise in the PR. I'm approving the current changes.

**Review by elena-zh — APPROVED** [review]

**Pull request merged**